lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -223,6 +419,18 @@ private static function get_block_nodes( $theme_json, $selectors = array() ) {
 						'path'     => array( 'styles', 'blocks', $name, 'elements', $element ),
 						'selector' => $selectors[ $name ]['elements'][ $element ],
 					);
+
+					// Handle any psuedo selectors for the element.
+					if ( isset( static::VALID_ELEMENT_PSEUDO_SELECTORS[ $element ] ) ) {
```

Comment:
That's beyond scope of this PR, but I wonder if `get_style_nodes` and `get_block_nodes` could be unified somehow – the logic is quite similar.

Reply:
Many of the functions in this class are become very large and not abiding by single responsibility principle. A refactor could be a good pairing exercise.